server/src/unifyfs_request_manager.c

```diff
@@ -1440,6 +1444,94 @@ static int process_unlink_rpc(reqmgr_thrd_t* reqmgr,
     return ret;
 }
 
+static int process_node_local_extents_get_rpc(reqmgr_thrd_t* reqmgr,
+                                              client_rpc_req_t* req)
+{
+    int ret = UNIFYFS_SUCCESS;
+    unifyfs_node_local_extents_get_in_t* in = req->input;
+    assert(in != NULL);
+    size_t num_req = in->num_req;
+    void* buffer = pull_margo_bulk_buffer(req->handle, in->bulk_data,
```

Comment:
You should pull the bulk buffer in the RPC handler (`unifyfs_node_local_extents_get_rpc()`) so that the transfer happens in a lightweight argothread. Follow the pattern from `unifyfs_mread_rpc()`, which is do the pull, then on success, assign `client_rpc_req_t.bulk_buf & .bulk_sz`

Reply:
Done